Mplus 2: Version 1.000; Mplus 1: Version 1.000

aaronbell opened the pull request:
438bea5: [gftools-packager] Mplus 1: Version 1.000; ttfautohint (v1.8.3) added

* Mplus 1 Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/8c6c65ff55f1725b9693b4c3f11d4c253fce8dd0.

 c5ae89a: [gftools-packager] Mplus 2: Version 1.000; ttfautohint (v1.8.3) added

* Mplus 2 Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/8c6c65ff55f1725b9693b4c3f11d4c253fce8dd0.

 e97a9da: [gftools-packager] Mplus 1 Code: Version 1.000; ttfautohint (v1.8.3) added

* Mplus 1 Code Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/8c6c65ff55f1725b9693b4c3f11d4c253fce8dd0.

 a29c278: [gftools-packager] Mplus Code Latin: Version 1.000; ttfautohint (v1.8.3) added

* Mplus Code Latin Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/8c6c65ff55f1725b9693b4c3f11d4c253fce8dd0.

aaronbell commented:
FYI @mintoming

aaronbell commented:
Regarding the FAILs on MplusCodeLatin:
- Font contains a width axis, and per the spec, it has been left without the widths being included in the instance names, causing this error. 
- Font metrics have been left to match the other Mplus CJK fonts. This way, if it is used in conjunction with any of the other ones, it will not cause and metrics issues. 
- Font is not monospace :)

aaronbell commented:
Thomas Linard has requested the addition of ExtraLight and SemiBold to the variable font STAT table:
https://github.com/coz-m/MPLUS_FONTS/issues/14

We are working to resolve upstream.

aaronbell commented:
Updated

## Mplus Code Latin: Version 1.000; ttfautohint (v1.8.3) added; Mplus 1 Code: Version 1.000; ttfautohint (v1.8.3) added; Mplus 1: Version 1.000; ttfautohint (v1.8.3) added; Mplus 2: Version 1.000; ttfautohint (v1.8.3) added

---
 25bb9a5: [gftools-packager] Mplus 1: Version 1.000; ttfautohint (v1.8.3) added

* Mplus 1 Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/3d3f039069da6b3dc8720c66b87b533481930e15.

 68194ea: [gftools-packager] Mplus 1 Code: Version 1.000; ttfautohint (v1.8.3) added

* Mplus 1 Code Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/3d3f039069da6b3dc8720c66b87b533481930e15.

 acfed76: [gftools-packager] Mplus 2: Version 1.000; ttfautohint (v1.8.3) added

* Mplus 2 Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/3d3f039069da6b3dc8720c66b87b533481930e15.

 c303d0d: [gftools-packager] Mplus Code Latin: Version 1.000; ttfautohint (v1.8.3) added

* Mplus Code Latin Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS at commit https://github.com/coz-m/MPLUS_FONTS/commit/3d3f039069da6b3dc8720c66b87b533481930e15.

aaronbell commented:
Now updated with two extra weights in each family

aaronbell commented:
I've dehinted the fonts with Kanji to save space.

davelab6 commented on ofl/mplus1/METADATA.pb:
```diff
@@ -0,0 +1,24 @@
+name: "Mplus 1"
+designer: "MPlus Fonts"
```
This should match the attribution on the existing M PLUS fonts (eg https://fonts.google.com/specimen/M+PLUS+Rounded+1c)

davelab6 commented on ofl/mplus1/METADATA.pb:
```diff
@@ -0,0 +1,24 @@
+name: "Mplus 1"
```
This should match the naming used already, eg https://fonts.google.com/specimen/M+PLUS+Rounded+1c

aaronbell replied:
Ah! For some reason I hadn't realized there was an extant MPLUS. I'll match it.

davelab6 reviewed (changes requested):
Looking forwards to this :) However, the few minor things I commented on will, I guess, require a rebuild and coordination with the upstream to propagate them back there, so please work with @mintoming on this :)

aaronbell commented:
@davelab6 These appear to be primarily metadata.pb changes rather than modifications to the actual font files. Internally, M PLUS 1P uses the "Mplus1p" name:

<img width="759" alt="Screen Shot 2021-06-15 at 8 33 29 AM" src="https://user-images.githubusercontent.com/8460297/122081937-65c2ba80-cdb4-11eb-8eb0-49bea0458c85.png">

I do need to rebuild the Code version as Coji's updated it, but aside from that, I don't think we need to rebuild the others.

aaronbell commented:
Updated

## M PLUS 1 Code: Version 1.000 added; M PLUS 2: Version 1.000 added; M PLUS 1: Version 1.000 added; M PLUS Code Latin: Version 1.000; ttfautohint (v1.8.3) added

---
 bd10172: [gftools-packager] M PLUS 1: Version 1.000 added

* M PLUS 1 Version 1.000 taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

 7b805a9: [gftools-packager] M PLUS 1 Code: Version 1.000 added

* M PLUS 1 Code Version 1.000 taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

 bfc48c4: [gftools-packager] M PLUS 2: Version 1.000 added

* M PLUS 2 Version 1.000 taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

 304e509: [gftools-packager] M PLUS Code Latin: Version 1.000; ttfautohint (v1.8.3) added

* M PLUS Code Latin Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

aaronbell commented:
Updated

## M PLUS 1: Version 1.000 added; M PLUS 2: Version 1.000 added; M PLUS 1 Code: Version 1.000 added; M PLUS Code Latin: Version 1.000; ttfautohint (v1.8.3) added

---
 bd10172: [gftools-packager] M PLUS 1: Version 1.000 added

* M PLUS 1 Version 1.000 taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

 7b805a9: [gftools-packager] M PLUS 1 Code: Version 1.000 added

* M PLUS 1 Code Version 1.000 taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

 bfc48c4: [gftools-packager] M PLUS 2: Version 1.000 added

* M PLUS 2 Version 1.000 taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

 304e509: [gftools-packager] M PLUS Code Latin: Version 1.000; ttfautohint (v1.8.3) added

* M PLUS Code Latin Version 1.000; ttfautohint (v1.8.3) taken from the upstream repo https://github.com/coz-m/MPLUS_FONTS.git at commit https://github.com/coz-m/MPLUS_FONTS/commit/8ef61dcea729781209fe08aedbd4e3cd23695469.

aaronbell commented:
Had some issues with conflicts between my local version of this and the repro :/. Think I got it all sorted out. Anyway, ended up rebuilding all the fonts to implement the fix Coji requested, and have updated metadata so this should be good to go now.

RosaWagner commented:
There are intriguing FAILs about duplicated instances and naming inconsistencies.

I wonder if we shouldn't update the v-metrics of the already published fonts, instead of adding new bad ones

aaronbell commented:
I can give a bit more clarity about the FAILS on the MPlus Latin font. 

1) It contains a width axis, which is what is leading to the "duplicated instances" (since GF doesn't allow for the width axis to be present in the fvar table). 

2) All of these are following the CJK vertical metrics guidelines as they're part of the same family. So while font bakery calls these FAILS, they're OK as is. 

3) I'll check the naming inconsistencies. I recall there was some confusion as MPlus was already on GF, so I needed to snap to the naming convention already established. Might have missed these.

RosaWagner commented:
>It contains a width axis, which is what is leading to the "duplicated instances" (since GF doesn't allow for the width axis to be present in the fvar table).

Indeed, the width=125 instances shouldn't be in the FVAR at all

davelab6 commented:
The `ofl/mplus1/static/` folder and its contents do not need to exist here, since they are not hand hinted.

davelab6 commented:
> Font metrics have been left to match the other Mplus CJK fonts. This way, if it is used in conjunction with any of the other ones, it will not cause and metrics issues.

I discussed this with @RosaWagner and @m4rc1e and we agreed this makes sense. 

> @davelab6 These appear to be primarily metadata.pb changes rather than modifications to the actual font files. Internally, M PLUS 1P uses the "Mplus1p" name:

Thanks for fixing this throughout, I just hacked it quickly in 2016 and didn't change the files internals.

RosaWagner commented:
@aaronbell I still don't have a working gftools[qa], can you send me a zip ?
also, should the 1p font be deprecated after this goes to production?

aaronbell commented:
1P is a rounded version which is different than these, so it shouldn't be deprecated. That said, it would probably be worth investigating if a new version of 1P should also be produced. 

For QA, are you just looking for fontbakery? Or something else?

RosaWagner commented:
Also browser previews; I can still generate fontbakery report, but I can't have browser screenshots to check consistency across platform and eventual visual weirdness.
